python/pyiceberg/table/__init__.py

```diff
@@ -225,17 +221,13 @@ def commit_transaction(self) -> Table:
         """
         # Strip the catalog name
         if len(self._updates) > 0:
-            response = self._table.catalog._commit_table(  # pylint: disable=W0212
+            self._table._do_commit(  # pylint: disable=W0212
```

Comment:
I'm fine with this being private for now, but we may want to actually expose it later.